### PR TITLE
Bulk actions framework

### LINKED ISSUE #136: Bulk Descriptive Metadata Download
1. from a list of Druids, fetch and output the MODS from the descMetadata stream
     - as a single \<xmlDocs> document (see https://consul.stanford.edu/display/chimera/MODS+bulk+loading)
2. Offline process managed as background task and tracked, delivered via a history screen similar to that developed for Replayable Spreadsheets 
3. Authorization checked for access with "Not authorized" error" on a per object basis
3. First instance of data output in bulk ... offer "druids from search" and "paste a druid list" choice like Bulk Update
4. Change name of "Bulk Update" to "Bulk actions" ... or implement separate "Bulk output" for this and (future) tracking sheet generation.

### PR DESCRIPTION
**Pull request by mejackreed**
Closes #136 

This pull request lays the foundation for the future of Argo's bulk actions by adding a framework and implementing descriptive metadata download.

screen shot 2016-02-08 at 5 21 52 pm
screen shot 2016-02-08 at 5 21 21 pm
screen shot 2016-02-08 at 5 21 38 pm

### CODE REVIEW

File: app/jobs/generic_job.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+##
```
Comment (mejackreed) [inline app/jobs/generic_job.rb]: I think this is a miscue by Hound.

**Comment by drh-stanford** [comment]
Re: schema. Since the `action_type` and `status` columns both seem to be enumerated types, it'd be nice to see somewhere what all the possible values are for them. Ideally with a check constraint on the schema.

**Comment by mejackreed** [comment]
Thanks for the comment @drh-stanford . In 8db6f29 I've added the validation for `action_type` at the model level and added a test for this. Is that sufficient?

**Comment by drh-stanford** [comment]
What about `status`? Or do you never query on status?

**Comment by mejackreed** [comment]
Ah.. yeah.. we don't query by status but we do check if "Completed". Hmm.. I feel like that is ok with being left more free form as we aren't executing any user provided params here. Thoughts?

**Comment by drh-stanford** [comment]
If it's meant to be free-form text then no need to validate it. :+1:

**Comment by drh-stanford** [comment]
I'm :+1: on this

**Pull request merged**